test/unit/spec/nbextensions/bulkImportCell/util-spec.js

```diff
@@ -0,0 +1,264 @@
+
+    const testFileType = 'someFileType';
+    const testAppId = 'kb_uploadmethods/import_fastq_sra_as_reads_from_staging';
+    const filePathIds = ['fastq_fwd_staging_file_name', 'fastq_rev_staging_file_name', 'name'];
+    const paramIds = [
+        'import_type',
+        'sequencing_tech',
+        'single_genome',
+        'interleaved',
+        'read_orientation_outward',
+        'insert_size_std_dev',
+        'insert_size_mean',
+    ];
+    // all of these tests use the 'kb_uploadmethods/import_fastq_sra_as_reads_from_staging'
+    // spec as given in test/data/testAppObj
+    /**
+     * This builds a quick and dirty test "model" around one or more instances of the same app
+     * spec. It just gets used multiple times (we're testing whether the model validates, not the
+     * various apps).
+     * @param {Array} inputs - an object where key is the fileType, and value is an array of file names
+     * @returns {Object} model - a Props object with data matching the file names and parameters
+     *  from the test app spec
+     */
+    function buildModel(fileInputs) {
+        const fileTypes = Object.keys(fileInputs);
+        // set up inputs
+        const params = {
+                import_type: 'SRA',
+                insert_size_mean: null,
+                insert_size_std_dev: null,
+                interleaved: 0,
+                read_orientation_outward: 0,
+                sequencing_tech: 'Illumina',
+                single_genome: 1,
+            },
+            inputs = {},
+            appParams = {},
+            fileParamIds = {},
+            otherParamIds = {};
+        fileTypes.forEach((fileType) => {
+            const files = fileInputs[fileType];
+            fileParamIds[fileType] = Object.assign([], filePathIds);
+            otherParamIds[fileType] = Object.assign([], paramIds);
+            inputs[fileType] = {
+                appId: testAppId,
+                files,
+            };
+            const filePaths = files.map((filename) => {
+                return {
+                    fastq_fwd_staging_file_name: filename,
+                    fastq_rev_staging_file_name: filename,
+                    name: filename,
+                };
+            });
+            appParams[fileType] = {
+                filePaths,
+                params: Object.assign({}, params),
+            };
+        });
+        return Props.make({
+            data: {
+                app: { fileParamIds, otherParamIds },
+                inputs,
+                params: appParams,
+            },
+            onUpdate: () => {},
+        });
+    }
+
+    describe('BulkImportCell Utility tests', () => {
+        let spec;
+        beforeAll(() => {
+            spec = Spec.make({ appSpec: TestAppObj.app.specs[testAppId] });
+        });
+
+        describe('evaluateAppConfig tests', () => {
+            // happy path tests with >= 1 file path sets
+            [1, 2, 5 /* 3, sir */].forEach((fileCount) => {
```

Comment:
remove commented-out data?

Reply:
Was more of a really bad joke, but done. 😅